doc/source/installation/apt_repo.rst

```diff
@@ -146,6 +145,95 @@ This will install all the packages from the bundle. Another option is to downloa
 
   When installing packages manually like this, you'll need to make sure to resolve all the dependencies and install missing packages yourself. Following packages will need to be installed before you can manually install Percona Server: ``mysql-common``, ``libjemalloc1``, ``libaio1`` and ``libmecab2``
 
+.. rubric:: AppArmor settings
+
+AppArmor is a kernel-integrated system which controls how applications access the file system by creating application profiles. If the installation of MySQL adds an AppArmor profile, the profile is in the following locations:
+
+* /etc/apparmor.d/usr.sbin.mysqld
+* /etc/apparmor.d/local/usr.sbin.mysqld
+
+The ``local`` version is empty except for comments. Add any changes specific for the server to the ``local`` file. 
+
+The ``usr.sbin.mysqld`` file has the following settings:
+
+.. sourcecode:: text
+
+    #include <tunables/global>
+
+    /usr/sbin/mysqld {
+      ...
+      # Allow data dir access
+      /var/lib/mysql/ r,
+      /var/lib/mysql/** rwk,
+
+      # Allow data files dir access
+        /var/lib/mysql-files/ r,
+        /var/lib/mysql-files/** rwk,
+
+      # Allow keyring dir access
+        /var/lib/mysql-keyring/ r,
+        /var/lib/mysql-keyring/** rwk,
+
+      # Allow log file access
+        /var/log/mysql/ r,
+        /var/log/mysql/** rw,
+      ...
+    }
+
+The settings govern how the files are accessed. For example, the data file directory access gives read (r) access to a directory and read, write, and lock access (rwk) to all directories and files underneath ``/mysql/``.
+
+You should download the `apparmor-utils` package when you are working with existing AppArmor profiles. The utilities allow you to edit a profile without stopping AppArmor or removing the profile.
+
+Before you edit a profile, you should change the profile to ``complain`` mode:
+
+.. sourcecode:: bash
+
+      # aa-complain /usr/sbin/mysqld
+      setting /usr/sbin/mysqld to complain mode
+
+In complain mode, you can edit the profile to add settings because you have relocated the data directory to a new location: ``/<volume>/dev/percona/data``:
```

Comment:
In complain mode, you can edit the profile to add settings because you have relocated the data directory:

Reply:
Fixed